prov/gni/include/gnix_freelist.h

```diff
@@ -123,6 +126,12 @@ static inline int _gnix_fl_alloc(struct dlist_entry **e, struct gnix_freelist *f
         fastlock_acquire(&fl->lock);
 
     if (dlist_empty(&fl->freelist)) {
+
+        if (fl->refill_size == 0) {
+            ret = -FI_ECANCELED;
```

Comment:
I believe `FI_ECANCELED` is the correct error code to use here since the allocation operation is canceled due to the given freelist being initialized with a refill size of zero. I will not change the error code unless anyone has objections.

I could add a warning message like "Unable to grow freelist(%p) since the refill size is 0" when `-FI_ECANCELED` is returned from `_gnix_fl_alloc`.